patches/tModLoader/Terraria/ModLoader/ModBiome.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Xna.Framework;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// This class represents a biome added by a mod. It exists to centralize various biome related hooks, handling a lot of biome boilerplate.
+	/// </summary>
+	public abstract class ModBiome : ModAVFX
+	{
+		// Basic Biome information
+		/// <summary>
+		/// Whether or not this biome impacts NPC shop prices.
+		/// </summary>
+		public virtual bool IsPrimaryBiome => false;
+		public override AVFXPriority Priority => AVFXPriority.BiomeLow;
+		public override int Music => 0;
+
+		internal int ZeroIndexType => Type - Loaders.Biomes.vanillaCount; 
+
+		// Bestiary properties
+		/// <summary>
+		/// The display name for this biome in the bestiary.
+		/// </summary>
+		public ModTranslation DisplayName { get; private set; }
+		/// <summary>
+		/// The path to the 30x30 texture that will appear for this biome in the bestiary.
+		/// </summary>
+		public virtual string BestiaryIcon => null;
+		/// <summary>
+		/// The path to the background texture that will appear for this biome behind npcs in the bestiary.
+		/// </summary>
+		public virtual string BackgroundPath => null;
+		/// <summary>
+		/// The color of the bestary background.
+		/// </summary>
+		public virtual Color? BackgroundColor => null;
+
+		public GameContent.Bestiary.ModBiomeBestiaryInfoElement ModBiomeBestiaryInfoElement { get; internal set; }
+
+		protected sealed override void Register() {
+			Type = Loaders.Biomes.Register(this);
+			RegisterAVFX(this);
+
+			DisplayName = Mod.GetOrCreateTranslation($"Mods.{Mod.Name}.BiomeName.{Name}");
+
+			ModBiomeBestiaryInfoElement = new GameContent.Bestiary.ModBiomeBestiaryInfoElement(Mod, DisplayName.Key, BestiaryIcon, BackgroundPath, BackgroundColor);
+		}
+
+		public sealed override void SetupContent() {
+			SetStaticDefaults();
+		}
+
+		public sealed override bool IsActive(Player player) => player.modBiomeFlags[ZeroIndexType];
+
+		/// <summary>
+		/// This is where you can set values for DisplayName.
+		/// </summary>
+		public virtual void SetStaticDefaults() {
+		}
+
+		/// <summary>
+		/// Return true if the player is in the biome.
+		/// </summary>
+		/// <returns></returns>
+		public virtual bool IsBiomeActive(Player player) => false;
```

Comment:
should be distinguished from `IsActive` more easily. Perhaps this could be protected, and have an internal accessor for tML only? Something like `CalculateIsPlayerInBiome`?